### ISSUE TITLE
Will you share pre-training code of StructBERT

### ISSUE THREAD

**Issue by kaansonmezoz** [alibaba/AliceMind#7]
Hi, I'm trying to code StructBERT from scratch. But I couldn't find any code examples pre-training about StructBERT. 
In the repository I've found codes for fine-tuning based on various datasets. 

Are you planning to share pre-training model's code for StructBert such as BertForPretraining in Transformers library ?

Thanks in advance 🙂

**Comment by wangwei7175878** [comment]
We currently have no plan to open source pre-training code, please refer to the paper for more details.

**Comment by kaansonmezoz** [comment]
Thanks for the clarification @wangwei7175878 . 

I've tried to implement my own version of Word Structural Objective (WSO). 
For me there were some unclear parts in the paper, so I had to come-up with my own solutions. 

Could you please explain more about the following questions ?

- Is it possible for a sentence to have more than one trigrams ?
- Since WSO is a single-sentence task, did you apply token shuffling to real single sentence or sequence of sentences ? Does it have to be an actual linguistic sentence ?
- Did you use same head for MLM and WSO (i.e Huggingface's BertOnlyMLMHead)? Since MLM and WSO are predicting tokens in the same way with different labels, MLM takes labels with masked tokens and WSO takes labels with shuffled tokens, same head can be used to predict token ids.

**Comment by wangwei7175878** [comment]
1. Yes, there can be more than one trigrams. Section 2.4: "5% of trigrams are selected for random shuffling"
2. We calculate WSO loss from sequence of sentences.
3. Yes, we use same head for MLM and WSO

**Comment by kaansonmezoz** [comment]
Thanks for the explanation @wangwei7175878

**Comment by luofuli** [comment]
Close this issue since it has been solved.

**Comment by Rachel-Yeah-Lee** [comment]
Hi @kaansonmezoz , I am tring to implement the Word Structural Objective as well, would you mind sharing the code with me, please?
